### ISSUE TITLE
PMA0015: Making Melong Think !!

### ISSUE THREAD

**Issue by TenzinGayche** [OpenPecha/think_and_translate#1]
# Introduction 
Monlam Melong, a prominent translation tool, aims to push the boundaries of this capability by introducing a key enhancement: the ability to "think" before translating. Inspired by recent advances in large language model (LLM) research, this project seeks to enable Monlam Melong to generate internal thoughts before producing final translations. By fostering a structured reasoning process, this initiative intends to improve the contextual accuracy, logical coherence, and overall quality of translations. The ability to "think" before translating is expected to significantly enhance Monlam Melong's performance, especially for nuanced and context-dependent translations. This document outlines the objectives, to-do list, risks, and milestones associated with this ambitious endeavor.

# To dos:

- [x] Create the data 
- [x] Add thoughts 
- [x] Fine Tunning Model for thought generation and Direct Translation
- [x]  Evaluate without optimisation 
- [x]  Ranking outputs for DPO
- [x]  Dpo Training 
  - [x] Dpo 1 new training dataset
  - [x] Dpo 2 new training dataset
- [x] Test the script
- [x]  Evaluate 

@kaldan007 

[Card link](https://github.com/orgs/OpenPecha/projects/107?pane=issue&itemId=91865719&issue=OpenPecha%7Cthink_and_translate%7C2)

**Comment by TenzinGayche** [comment]
we got 33k high quality data with the latest script, the last one got only 4k

**Comment by TenzinGayche** [comment]
creating new set of subdata that doesnt do just analysis

**Comment by TenzinGayche** [comment]
Evaluation for the 2 model with 5 checkpoint of BLEU,TER,BLEURT is being done

**Comment by TenzinGayche** [comment]
![Image](https://github.com/user-attachments/assets/fffaa121-f08b-4a1d-bc20-c38bc8315ffc)

![Image](https://github.com/user-attachments/assets/ab53f75a-359b-486e-a8c9-a9253f204fde)

**Comment by TenzinGayche** [comment]
![Image](https://github.com/user-attachments/assets/31df4b15-c3b9-4e3e-8cdc-43b59ad05313)

**Comment by gangagyatso4364** [comment]
Ranking work pending, will assign the task

**Comment by TenzinGayche** [comment]
![Image](https://github.com/user-attachments/assets/6b0b0cad-e939-4fe7-9e79-16fa76a5105e)
Overfitting

**Comment by TenzinGayche** [comment]
![Image](https://github.com/user-attachments/assets/4f76e793-bb43-4c88-95fd-a4f75562d604)

![Image](https://github.com/user-attachments/assets/80f7a779-d2b8-4814-b12f-27e4024bf629)

**Comment by TenzinGayche** [comment]
![Image](https://github.com/user-attachments/assets/a78c8a3b-5eaa-4042-8f52-0724eb31e546)

**Comment by TenzinGayche** [comment]
![Image](https://github.com/user-attachments/assets/01bcb9e7-d873-4608-b533-1dcb4c45527d)
